ipinfo/client.go

```diff
@@ -64,21 +65,46 @@ func NewClient(
 	}
 }
 
+// newRequest for IPV4
+func (c *Client) newRequest(
+	ctx context.Context,
+	method string,
+	urlStr string,
+	body io.Reader,
+) (*http.Request, error) {
+	return c.newRequestbase(ctx, method, urlStr, body, false)
+}
+
+// newRequest for IPV6
+func (c *Client) newRequestV6(
+	ctx context.Context,
+	method string,
+	urlStr string,
+	body io.Reader,
+) (*http.Request, error) {
+	return c.newRequestbase(ctx, method, urlStr, body, true)
+}
+
 // `newRequest` creates an API request. A relative URL can be provided in
 // urlStr, in which case it is resolved relative to the BaseURL of the Client.
 // Relative URLs should always be specified without a preceding slash.
-func (c *Client) newRequest(
+func (c *Client) newRequestbase(
 	ctx context.Context,
 	method string,
 	urlStr string,
 	body io.Reader,
+	useIPv6 bool,
 ) (*http.Request, error) {
 	if ctx == nil {
 		ctx = context.Background()
 	}
 
 	u := new(url.URL)
 
+	if useIPv6 {
+		c.BaseURL, _ = url.Parse(defaultBaseURLIPv6)
```

Comment:
This updates it for all uses now and later. Needs to be temporary just for this request.